builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -147,24 +147,24 @@ func resourceAwsS3Bucket() *schema.Resource {
 			},
 
 			"versioning": {
-				Type:     schema.TypeSet,
+				Type:     schema.TypeList,
```

Comment:
Technically yes, as we are moving from the hashcode to an int - but I figured as it was a 0.8 release and we only ever allow 1 set of versioning, that we wouldn't cause any issue here